cmd/deploy.go

```diff
@@ -241,6 +241,11 @@ func autofillApimodel(dc *deployCmd) {
 				Secret:   secret,
 				ObjectID: servicePrincipalObjectID,
 			}
+		} else if dc.containerService.Properties.ServicePrincipalProfile == nil && dc.ClientID.String() != "" && dc.ClientSecret != "" {
```

Comment:
what about the case where `dc.containerService.Properties.ServicePrincipalProfile` is not nil but is empty? For example, ``` "servicePrincipalProfile": {
      "clientId": "",
      "secret": ""
    }```

Reply:
We need to address this case since it applies to most example json apimodel in the `examples/` folder